### PR TITLE
Performance

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: frontend/src/App.vue [hunk 1]
```diff
@@ -104,48 +107,57 @@ export default {
...
-              question.answer = response
-            })
-          }
-          if (question.validate === FUNCTION) {
-            this.rpc.invoke("evaluateMethod", [[question.answer, curentPromptAnswers], questionName, "validate"]).then(response => {
-              question.isValid = (_.isString(response) ? false : response)
-              question.validationMessage = (_.isString(response) ? response : undefined)
-            })
+          
+          if (!isEqual) {
```
Comment (tomer-epstein) [inline frontend/src/App.vue]: conside check (question.isWhen !==false)

Comment (slavik-lvovsky) [reply frontend/src/App.vue]: In the inquirer js documentation they do not tell that when "when ===false" all other question parameters do not change. 
We might create a bug applying such an assumption.

Comment (tomer-epstein) [reply frontend/src/App.vue]: In the yo cli if when === false then question value is undefined.

File: frontend/src/App.vue [hunk 2]
```diff
@@ -104,54 +107,71 @@ export default {
...
+            if (question.when === FUNCTION) {
+              that.rpc.invoke("evaluateMethod", [[newAnswers], questionName, "when"]).then(response => {
+                question.isWhen = response
+                if (question.isWhen === true) {
+                  that.updateQuestion(question, questionName, newAnswers)
+                }
+              })
+            } else if (question.isWhen === true) {
+              that.updateQuestion(question, questionName, newAnswers)
+            }
```
Comment (tomer-epstein) [inline frontend/src/App.vue]: should be if (

File: frontend/src/App.vue [hunk 3]
```diff
@@ -104,54 +107,71 @@ export default {
...
           }
         })
       }
     }
   },
   methods: {
+    updateQuestion(question, questionName, newAnswers) {
+      if (question._default === FUNCTION) {
```
Comment (tomer-epstein) [inline frontend/src/App.vue]: honestelly, I prefer to see all those "if"s under if (question.isWhen === true)
no need to wrap with a function here.

File: frontend/src/App.vue (hunk 2 above)
Comment (tomer-epstein) [inline frontend/src/App.vue]: place it in one function -> that.updateQuestion(question, questionName, newAnswers, (question.when === FUNCTION))

File: frontend/src/App.vue [hunk 4]
```diff
@@ -104,54 +107,68 @@ export default {
...
-            })
-          }
-          if (question.validate === FUNCTION) {
-            this.rpc.invoke("evaluateMethod", [[question.answer, curentPromptAnswers], questionName, "validate"]).then(response => {
-              question.isValid = (_.isString(response) ? false : response)
-              question.validationMessage = (_.isString(response) ? response : undefined)
-            })
+          
+          if (!isEqual) {
+            await that.updateQuestion(question, questionName, newAnswers)
```
[the remark this answers is not in the document]
Comment (slavik-lvovsky) [reply frontend/src/App.vue]: it is used twice in if (isEqual) block that is why i created it

File: frontend/src/App.vue [hunk 5]
```diff
@@ -104,54 +107,68 @@ export default {
...
+          if (!isEqual) {
+            await that.updateQuestion(question, questionName, newAnswers)  
           }
         })
       }
     }
   },
   methods: {
+    async updateQuestion(question, questionName, newAnswers) {
+      if (question.when === FUNCTION) {
```
[the remark this answers is not in the document]
Comment (slavik-lvovsky) [reply frontend/src/App.vue]: it is used 5 times in the function that is why i decided to get it from outside where it is already defined

File: frontend/src/App.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (tomer-epstein) [reply frontend/src/App.vue]: can place this line inside if (isEqual)
          const questionName = question.name
or can use question.name instead

**Pull request merged**